graphql_compiler/query_formatting/common.py

```diff
@@ -28,6 +28,63 @@ def _raise_invalid_type_error(name, expected_python_type_name, value):
                                                          type(value).__name__))
 
 
+def _deserialize_anonymous_json_argument(expected_type, value):
+    """Deserialize argument. See docstring of deserialize_json_argument."""
+    allowed_types_for_graphql_type = {
+        GraphQLDate: (six.string_types,),
+        GraphQLDateTime: (six.string_types,),
+        GraphQLFloat: (six.string_types, float, six.integer_types),
+        GraphQLDecimal: (six.string_types, float, six.integer_types),
+        GraphQLInt: (six.integer_types,),
+        GraphQLString: (six.string_types,),
+        GraphQLBoolean: (bool,),
+        GraphQLID: (six.string_types,),
+    }
+
+    # Check if the type of the value is correct
+    correct_type = True
+    expected_python_types = allowed_types_for_graphql_type[expected_type]
+    if isinstance(value, bool) and not GraphQLBoolean.is_same_type(expected_type):
+        correct_type = False  # We explicitly disallow passing boolean values for non-boolean types
+    if not isinstance(value, expected_python_types):
+        correct_type = False
+    if not correct_type:
+        raise ValueError(u'Unexpected type {}. Expected one of {}.'
+                         .format(type(value), expected_python_types))
+
+    # Use the default GraphQL parser to parse the value
+    return expected_type.parse_value(value)
```

Comment:
Per #623, it seems like this line could throw an `arrow` error which I believe might not be a subclass of `ValueError`, so this code might have unexpected behavior.